packages/compartment-mapper/src/import-hook.js

```diff
@@ -144,6 +146,10 @@ export const makeImportHookMaker = (
       const candidates = [];
       if (moduleSpecifier === '.') {
         candidates.push('./index.js');
+        // https://github.com/endojs/endo/issues/1363
+        if (packageMainExport) {
+          candidates.push(packageMainExport);
+        }
```

Comment:
I think that, if there is a `main` in `package.json`, it should be used instead of `index.js`, not as a fallback when `index.js` does not exist. We should have tests, and it would be good for that test to verify parity with Node.js.